### ISSUE TITLE
Handling providedCompile/Runtime

### ISSUE THREAD

**Issue by CSBaum** [nebula-plugins/gradle-lint-plugin#21]
I noticed that lint is flagging my providedCompile dependencies as needing to move to the compile configuration but they are technically extensions of compile (if i understand gradle's configuration bits correctly).  

```
warning   unused-dependency                  this dependency should be moved to configuration compile
build.gradle:152
providedCompile ('javax.servlet:javax.servlet-api:3.+') {transitive = false}
```
If you need any output please let me know.

Thanks in advance.

Sean

**Comment by jkschneider** [comment]
Thanks again, @CSBaum. Fixed with 1ecc460733ee3524f2e14c4e686e986e25c88d0a and released as 0.26.0.
